### PR TITLE
Code quality improvement

### PR DESCRIPTION
**Pull request by vivekscl**
* Refactor many areas of the code that I wrote
* Clear several TODOs
* Add more tests
* Update collated folders
@okkhoy Ready for Review

### CODE REVIEW

File: collated/main/vivekscl.md [hunk 1]
```diff
@@ -80,20 +77,31 @@
...
-            newTags.add(toAdd);
-            newPerson.setTags(newTags);
+            Person newPerson = setTagsForNewPerson(oldPerson, toAdd, true);
 
             addressBook.updatePerson(oldPerson, newPerson);
             indicateAddressBookChanged();
         }
     }
 
+    private Person setTagsForNewPerson(ReadOnlyPerson oldPerson, Tag tagToAddOrRemove, boolean isAdd) {
```
Comment (okkhoy) [inline collated/main/vivekscl.md]: header comments?

File: collated/main/vivekscl.md [hunk 2]
```diff
@@ -164,6 +172,168 @@
...
+ * Since all methods and attributes are static, an instance of this class will not be made.
+ * Hence, the class is set to final with a default private constructor to prevent accidental creation of this class.
+ * Guarantees: immutable; window size is valid as declared in {@link #isValidWindowSize(String)}
+ */
+public final class WindowSize {
+
+    public static final String MESSAGE_WINDOW_SIZE_CONSTRAINTS = "The only allowed sizes are small, med and big";
+
+    // Commonly used sizes
+    public static final double DEFAULT_HEIGHT = 600;
```
Comment (okkhoy) [inline collated/main/vivekscl.md:185]: can extract these as a class if you wish

File: collated/main/vivekscl.md [hunk 3]
```diff
@@ -164,6 +172,168 @@
...
+            allPersonsString += "\n" + personIterator.next() + "\n";
+        }
+        if (tagIterator.hasNext()) {
+            allTagsString += "\n" + tagIterator.next();
+        }
+
+        while (personIterator.hasNext()) {
+            allPersonsString = allPersonsString.concat(personIterator.next() + "\n");
+        }
+        while (tagIterator.hasNext()) {
```
Comment (okkhoy) [inline collated/main/vivekscl.md]: spacing

File: collated/main/vivekscl.md [hunk 4]
```diff
@@ -164,6 +172,168 @@
...
+    public String toString() {
+        Iterator<Person> personIterator = persons.iterator();
+        Iterator<Tag> tagIterator = tags.iterator();
+        String allPersonsString = "Persons: ";
+        String allTagsString = "Tags: ";
+
+        if (personIterator.hasNext()) {
+            allPersonsString += "\n" + personIterator.next() + "\n";
+        }
+        if (tagIterator.hasNext()) {
```
Comment (okkhoy) [inline collated/main/vivekscl.md]: spacing

Comment (vivekscl) [reply collated/main/vivekscl.md]: Spacing as in, add a space between the if statements?

**Pull request merged**